packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -112,7 +117,7 @@ const useUtilityClasses = (ownerState: PickersDayProps<any>) => {
   const slots = {
     root: [
       'root',
-      selected && 'selected',
+      selected && (!outsideCurrentMonth || showDaysOutsideCurrentMonth) && 'selected',
```

Comment:
Correctly fix this case

![Screenshot 2023-05-31 at 17 40 34](https://github.com/mui/mui-x/assets/4941090/a56ec102-6ca7-4b89-a85f-a54c19f8132b)